### PR TITLE
JP-2526: Fix kdtree memory issues on Linux systems

### PR DESCRIPTION
**Pull request by larrybradley**
Resolves [JP-2526](https://jira.stsci.edu/browse/JP-2526)

**Description**

This PR fixes the KDTree calculation to use only non-finite source positions to prevent memory issues on Linux systems. It also filters a RuntimeWarning for non-finite source positions.

Checklist
- [ ] Tests
- [ ] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

**Comment by larrybradley** [comment]
For non-finite source positions, `nn_dist` is assigned to `np.nan` and `nn_label` is assigned to -1.  I chose -1 to keep the `nn_label` as `int` type (inserting `np.nan` requires float type).

File: CHANGES.rst [hunk 1]
```diff
@@ -97,6 +95,12 @@ skymatch
 
 - Updated to populate the "BKGMETH" keyword in output files. [#6736]
 
+source_catalog
+--------------
+
+- Fixed the KDTree calculation to use only non-finite source positions
```
Comment (hbushouse) [inline CHANGES.rst:101]: I think you mean *finite* here! We're NOT using the non-finite positions.

**Review by hbushouse — APPROVED** [review]
Other than the one minor change log comment, LGTM.

**Comment by hbushouse** [comment]
Hmmm, one of the new lines of code seems to be causing a CI failure.

File: CHANGES.rst (hunk 1 above)
Comment (larrybradley) [reply CHANGES.rst]: Thanks! 😃

**Comment by larrybradley** [comment]
The SDP failure doesn't seem related.  I'll investigate the other one.  The pipeline ran fine locally for me on the troublesome dataset.

**Pull request merged**